src/components/common/ProgressBar/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import styled from 'styled-components'
+
+export type Props = {
+  readonly percentage?: string
+  readonly activeColor?: string
+  readonly showLabel?: boolean
+}
+
+const Wrapper = styled.div<Props>`
+  display: flex;
+  align-items: center;
+
+  > div {
+    width: 16rem;
+    height: 0.6rem;
+    position: relative;
+    border-radius: 16rem;
+    background: ${({ theme }): string => theme.borderPrimary};
+  }
+
+  > div > span {
+    width: ${({ percentage }): string => percentage || '0'}%;
+    max-width: 100%;
+    height: 100%;
+    position: absolute;
+    left: 0;
+    top: 0;
+    background-color: ${({ activeColor, theme }): string => (activeColor ? activeColor : theme.green)};
+    border-radius: 16rem;
+  }
+
+  > b {
+    color: ${({ activeColor, theme }): string => (activeColor ? activeColor : theme.green)};
+    margin: 0 0 0 0.7rem;
+  }
+`
+
+export function ProgressBar(props: Props): JSX.Element {
+  const { percentage = '0', showLabel = true } = props
+
+  return (
+    <Wrapper {...props}>
+      <div>
```

Comment:
I kinda like having a compact styled component with very specific nested selectors (> b { ....)

But in terms of readability, would it be preferred to turn the nested elements into it's own styled components? E.g.

<div> --> OuterBar
<span> --> InnerBar

?

Reply:
I'm happy with both approaches. By the size of this component, your approach is good. 
Probably u are right that using  it's own styled components is more readable, but it's kind of fine in this specific case.

Reply:
I personally prefer smaller styled components, but am ok with this approach too.